src/engines/import_data_engine.py

```diff
@@ -59,9 +58,27 @@ def clear_import(self):
         self.state.last_algorithm = None
 
     def read_data(self, columns: Optional[List[str]] = None):
-        self.imported_data = self.reader_data.read(columns)
-        self.state.imported_data = self.imported_data
-        self.state.raw_data = self.imported_data.copy()
+        self.state.imported_data = self.reader_data.read(columns)
+        self.state.raw_data = self.state.imported_data.copy()
+        self.state.reduced_columns = []
+        self.state.steps_visualization = None
+        self.state.algorithm_results_widgets = {}
+        self.state.last_algorithm = None
+
+    def limit_data(self, columns: Optional[List[str]] = None, limit_type: Optional[str] = None, limit_num: Optional[str] = None):
+        self.drop_additional_columns()
+        if columns is not None:
+            self.state.raw_data = self.state.raw_data[columns]
+        if limit_type is not None:
+            if limit_type == "first":
+                self.state.raw_data = self.state.raw_data.iloc[:limit_num]
+            elif limit_type == "random":
+                self.state.raw_data = self.state.raw_data.sample(limit_num)
```

Comment:
I think we should reset index here (`.reset_index(drop=True)`), because after sampling data the index in the table is in weird order and as of now we don't set any specific column to be the index, so index has no meaning, so we won't lose data.

Reply:
Good idea